chd_test.go

```diff
@@ -50,7 +50,7 @@ func TestCHDBuilder(t *testing.T) {
 	}
 	c, err := b.Build()
 	assert.NoError(t, err)
-	assert.Equal(t, 7, len(c.keys))
+	assert.Equal(t, 15, len(c.keys))
```

Comment:
Previously `n=7` because it chose `n=len(keys)`. But per the paper, `m=len(keys)` and `n=m*c`. So `n` is no longer 7 (the number of input keys) but rather some multiple of the number of input keys--and I chose, based on rough guidance from the paper, `n=1+m*c` where `c=2`; thus `n=15`.

However, to be consistent about adhering to the paper, now that we're not setting `n=len(keys)`, I believe that `c.keys` is a misnomer now (or it may have been doing double duty holding both the keys and the vertices in the graph theoretical model of the CHD algorithm). So, this should be changed in the PR.

(Of course, all of this is subject to the caveat that I could be totally misunderstanding why you chose `n=len(keys)` not `m=len(keys)`.)